src/macos/observer.rs

```diff
@@ -155,31 +154,24 @@ impl Drop for ObserverRegistration {
     }
 }
 
-declare_class! {
-    pub(crate) struct Observer;
-
+define_class! {
     // SAFETY:
-    // * The superclass NSObject does not have any subclassing requirements.
-    // * Interior mutability is a safe default.
-    unsafe impl ClassType for Observer {
-        type Super = NSObject;
-        type Mutability = mutability::InteriorMutable;
-        const NAME: &'static str = "MundyObserver";
-    }
-
-    impl DeclaredClass for Observer {
-        type Ivars = Ivars;
-    }
+    // - The superclass NSObject does not have any subclassing requirements.
+    // - `MyCustomObject` does not implement `Drop`.
+    #[unsafe(super(NSObject))]
+    #[name = "MundyObserver"]
+    #[ivars = Ivars]
+    pub(crate) struct Observer;
```

Comment:
This is also nice!

Reply:
I'm also pretty happy that I was able to remove `unsafe impl Send` and `unsafe impl Sync` 🎉